FIXME

Not really sure if PHY_CONN_TYPE_EXT2_PHY is
required to be here. I didn't find any uses of
this throughout the code.
--- a/shared/opensource/boardparms/bcm963xx/boardparms_47189.c
+++ b/shared/opensource/boardparms/bcm963xx/boardparms_47189.c
@@ -125,6 +125,27 @@ static bp_elem_t g_bcm947189acr[] = {
   {bp_last}
 };
 
-bp_elem_t * g_BoardParms[] = {g_bcm947189ref, g_bcm947189ref2, g_bcm947189acnrm, g_bcm947189acnrh, g_bcm947189acdbmr, g_bcm947452eapl, g_bcm947189acr, 0};
+static bp_elem_t g_bcm947189cyrus[] = {
+  /* GPHY in RGMII 0, PLC in RGMII 1 */
+  {bp_cpBoardId,               .u.cp = "947189cyrus"},
+  {bp_ucPhyType0,              .u.uc = BP_ENET_NO_PHY},
+    {bp_ulPortMap,             .u.ul = 0x1},
+    {bp_ulPhyId0,              .u.ul = 24 | MAC_IF_RGMII | PHY_INTEGRATED_VALID | PHY_EXTERNAL | (ADVERTISE_ALL_GMII | PHY_ADV_CFG_VALID)},
+      {bp_usPhyConnType,       .u.us = PHY_CONN_TYPE_EXT_PHY},
+      {bp_ucPhyDevName,        .u.cp = "eth0"},
+      {bp_usGpioPhyReset,      .u.us = 0x02},
+  {bp_usGpioBitbangI2cScl,     .u.us = BP_GPIO_13_AL}, 
+  {bp_usGpioBitbangI2cSda,     .u.us = BP_GPIO_12_AL},
+  {bp_usGpioSpiClk,            .u.us = BP_GPIO_21_AH},
+  {bp_usGpioSpiCs,             .u.us = BP_GPIO_24_AH},
+  {bp_usGpioSpiMiso,           .u.us = BP_GPIO_22_AH},
+  {bp_usGpioSpiMosi,           .u.us = BP_GPIO_23_AH},
+  {bp_usGpioLedDriverEnable,   .u.us = BP_GPIO_11_AH},
+  {bp_usGpioLedFet,            .u.us = BP_GPIO_7_AH},
+  {bp_ulMemoryConfig,		   .u.ul = BP_DDR_DEVICE_WIDTH_16 | BP_DDR_HALF_TREF | BP_DDR_TRFC_160NS | BP_DDR_TOTAL_SIZE_256MB},
+  {bp_last}
+};
+
+bp_elem_t * g_BoardParms[] = {g_bcm947189ref, g_bcm947189ref2, g_bcm947189acnrm, g_bcm947189acnrh, g_bcm947189acdbmr, g_bcm947452eapl, g_bcm947189acr, g_bcm947189cyrus, 0};
 
 
--- a/shared/opensource/include/bcm963xx/boardparms.h
+++ b/shared/opensource/include/bcm963xx/boardparms.h
@@ -780,6 +780,7 @@ typedef struct _BP_MOCA_INFO {
 #define PHY_CONN_TYPE_PLC               6  /* switch MAC port connected to PLC */
 #define PHY_CONN_TYPE_FEMTO             7  /* switch MAC port connected to FEMTO */
 #define PHY_CONN_TYPE_MOCA_ETH          8  /* switch MAC port connected to MOCA ethernet */
+#define PHY_CONN_TYPE_EXT2_PHY          9  /* switch MAC port connected to external PHY */
 
 #define PHY_CONN_TYPE_NOT_DEFINED       0xffff  /* not specified */
 #define PHY_DEVNAME_NOT_DEFINED         0       /* not specified */
--- a/shared/opensource/drv/phys/phy_bp_parsing_47189.c
+++ b/shared/opensource/drv/phys/phy_bp_parsing_47189.c
@@ -75,6 +75,9 @@ phy_type_t bp_parse_phy_type(const EMAC_
     case PHY_CONN_TYPE_EXT_PHY:
         phy_type = PHY_TYPE_EXT1;
         break;
+    case PHY_CONN_TYPE_EXT2_PHY:
+        phy_type = PHY_TYPE_EXT2;
+        break;
     case PHY_CONN_TYPE_PLC:
         break;
     default:
